flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherTest.java

```diff
@@ -578,6 +579,45 @@ public void testFatalErrorIfRecoveredJobsCannotBeStarted() throws Exception {
         fatalErrorHandler.clearError();
     }
 
+    /** Tests that the {@link Dispatcher} do not fail by {@link JobFinishingException}. */
+    @Test
+    public void testNonFatalErrorWhenRecoveredJobFinishingException() throws Exception {
```

Comment:
Maybe we should add another test similar to `DispatcherTest.testJobDataAreCleanedUpInCorrectOrder` that ensures that we clean up things properly.

Reply:
I add another test case in `org.apache.flink.runtime.jobmaster.JobMasterServiceLeadershipRunnerTest#testJobAlreadyDone` I think this can cover the current case, what do you think ?